.pre-commit-config.yaml

```diff
@@ -0,0 +1,67 @@
+ci:
+  skip: [pyright]
+
+repos:
+-   repo: https://github.com/Zac-HD/shed
+    rev: 0.10.8
+    hooks:
+    -   id: shed
+        args: ['--py39-plus']
+
+-   repo: https://github.com/RobertCraigie/pyright-python
+    rev: v1.1.286
+    hooks:
+    -   id: pyright
+        entry: env PYRIGHT_PYTHON_FORCE_VERSION=latest pyright
+        args: ['--pythonversion=3.11', '--warnings']
```

Comment:
I forget the exact syntax here, but we'll want this to run on all-files rather than passing changed filenames, because changing one file can create typecheck errors in another unchanged file.

Reply:
In the CI check it will run on all files (it runs `pre-commit run pyright --all-files`), but I personally prefer the commit hook not to run on all files as that comes with a decently hefty time penalty (5->10 seconds on my machine).